sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -287,29 +325,29 @@ public int getNumShards() {
     /**
      * When a sink has requested windowed or triggered output, this method will be invoked to return
      * the file {@link ResourceId resource} to be created given the base output directory and a
-     * (possibly empty) extension from {@link FileBasedSink} configuration
-     * (e.g., {@link CompressionType}).
+     * {@link FileMetadataProvider} containing information about the file, including a suggested
+     * extension (e.g. coming from {@link CompressionType}).
```

Comment:
Yeah, it's used in FileBasedSink via the WritableByteChannel subclass. This subclassing was a bit awkward, but the easiest way to keep things backwards compatible I think.